ote/src/clj/ote/services/transport.clj

```diff
@@ -126,26 +126,31 @@
       (let [ckan (ckan/->CKAN (:api nap-config) (get-in user [:user :apikey]))
 
             ;; Insert to our database
-            {id ::t-operator/id :as operator}
-            (insert! db ::t-operator/transport-operator
-                     (dissoc data  ::t-operator/id :new?))
+            operator  (insert! db ::t-operator/transport-operator
+                        (dissoc data  ::t-operator/id :new?))
 
             ;; Create organization in CKAN
-            {ckan-group-id :ckan/id :as ckan-response}
-            (ckan/create-organization!
-             ckan
+            ckan-response (ckan/create-organization! ckan
              {:ckan/name (str "transport-operator-" (::t-operator/id operator))
               :ckan/title (::t-operator/name operator)})]
-
         ;; Update CKAN org id
         (update! db ::t-operator/transport-operator
-                 {::t-operator/ckan-group-id ckan-group-id}
-                 {::t-operator/id id})
+                 {::t-operator/ckan-group-id (get-in ckan-response [:ckan/result :ckan/id])} ;; Values to set
+                 {::t-operator/id (::t-operator/id operator)} ;; Where clause
+                 )
         operator))
 
-    (authorization/with-transport-operator-check
-      db user (::t-operator/id data)
-      #(upsert! db ::t-operator/transport-operator data))))
+    ;; Edit transport operator
+    (tx/with-transaction db
+       (let [ckan (ckan/->CKAN (:api nap-config) (get-in user [:user :apikey]))
+            operator (authorization/with-transport-operator-check
```

Comment:
The whole operation should be done within the transport operator check, not just the upsert.